File: php/factory/abstractfactory/Tyre.php
<?php

/**
 * 汽车座椅
 */

interface Tyre
{
    public function revolve();
}

/**
 * 高级车的座椅
 */
class LuxuryTyre implements Tyre {
    public function revolve()
    {
        echo '耐磨损';
    }
}


class LowTyre implements Tyre {
    public function revolve()
    {
        echo '磨损快';
    }
}

File: java/pattern/src/factory/abstractfactory/Tyre.java
package factory.abstractfactory;

public interface Tyre {
    void revolve();
}

class LuxuryTyre implements Tyre {

    @Override
    public void revolve() {
        System.out.println("耐磨损");
    }
}

class LowTyre implements Tyre {

    @Override
    public void revolve() {
        System.out.println("磨损快");
    }
}
